adapters/unruly/unruly.go

```diff
@@ -12,137 +12,148 @@ import (
+			return nil, "", errors
+		}
+		var unrulyExt openrtb_ext.ExtImpUnruly
+		//err = json.Unmarshal(bidderExt.Bidder, &unrulyExt)
+		//if err != nil {
+		if err := json.Unmarshal(bidderExt.Bidder, &unrulyExt); err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("siteid not provided in imp id=%s. Abort all Request", imp.ID),
 			}
-			errs = append(errs, errors...)
+			errors = append(errors, err)
+			return nil, "", errors
+		}
+		unrulyExtCopy, err := json.Marshal(&unrulyExt)
+		if err != nil {
+			errors = append(errors, err)
+			return nil, "", errors
+		}
+		bidderExtCopy := struct {
+			Bidder json.RawMessage `json:"bidder,omitempty"`
+		}{unrulyExtCopy}
+		impExtCopy, err := json.Marshal(&bidderExtCopy)
+		if err != nil {
+			errors = append(errors, err)
+			return nil, "", errors
+		}
+		imp.Ext = impExtCopy
+		req.Imp[i] = imp
+		if uri == "" {
+			uri = fmt.Sprintf("%s", a.endPoint)
 		}
 	}
-	return adapterRequests, errs
+	return req, uri, errors
 }
 
-func getMediaTypeForImpWithId(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
-	for _, imp := range imps {
-		if imp.ID == impID {
-			return openrtb_ext.BidTypeVideo, nil
-		}
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
 	}
-	return "", &errortypes.BadInput{
-		Message: fmt.Sprintf("Failed to find impression \"%s\" ", impID),
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
 	}
-}
 
-func CheckResponse(response *adapters.ResponseData) error {
 	if response.StatusCode != http.StatusOK {
-		return &errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
-		}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
 	}
-	return nil
-}
-
-func convertToAdapterBidResponse(response *adapters.ResponseData, internalRequest *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
-	var errs []error
 	var bidResp openrtb2.BidResponse
 	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
-		return nil, []error{err}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
 	}
+
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	var errs []error
 	for _, sb := range bidResp.SeatBid {
 		for i := range sb.Bid {
-			bidType, err := getMediaTypeForImpWithId(sb.Bid[i].ImpID, internalRequest.Imp)
+			var bla, err = getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp)
```

Comment:
Nitpick: Can we rename `bla` to `bidType` or something more descriptive?

Reply:
done